GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -124,39 +124,112 @@ public static bool TrySetRequiredGitConfigSettings(Enlistment enlistment)
+
+                // Ensure hooks path is configured correctly.
                 { "core.hookspath", expectedHooksPath },
+
+                // Hostname is no longer sufficent for VSTS authentication. VSTS now requires dev.azure.com/account to determine the tenant.
+                // By setting useHttpPath, credential managers will get the path which contains the account as the first parameter. They can then use this information for auth appropriately.
                 { GitConfigSetting.CredentialUseHttpPath, "true" },
                 { "credential.validate", "false" },
+
+                // This setting is not needed anymore, because current version of gvfs does not use index.lock.
+                // (This change was introduced initially to prevent `git diff` from acquiring index.lock file.)
                 { "diff.autoRefreshIndex", "true" },
+
+                // Turn off of git garbage collection. Git garbage collection does not work with virtualized object. Check with Stolee - probably not needed.
                 { "gc.auto", "0" },
+
+                // Prevent git GUI from displaying a GC warning. https://mseng.visualstudio.com/AzureDevOps/_git/GVFS/pullrequest/184856?_a=overview
                 { "gui.gcwarning", "false" },
+
+                // Update git settings to enable optimizations in git 2.20
+                // Set 'index.threads=true' to enable multi-threaded index reads
                 { "index.threads", "true" },
+
+                // gvfs index parser would work only with index versions 4 and above.
                 { "index.version", "4" },
+
+                // Turned off for performance reasons.
                 { "merge.stat", "false" },
+
+                // Update git version to 2.18.0.gvfs.1.26.gf61ade4 - This updates the git version to 2.18.0 which includes changes to the rename detection to use
+                // config setting which was previously turned off with a patch in git.
                 { "merge.renames", "false" },
+
+                // Don't use bitmaps to determine pack file contents, b'coz we use MIDX for this.
                 { "pack.useBitmaps", "false" },
+
+                // Update Git to include sparse push algorithm
                 { "pack.useSparse", "true" },
+
+                // Stop automatic git GC
                 { "receive.autogc", "false" },
+
+                // Update git settings to enable optimizations in git 2.20
+                // Set 'reset.quiet=true' to speed up 'git reset <foo>"
                 { "reset.quiet", "true" },
+
+                // Configure git to use our serialize status file
                 { "status.deserializePath", gitStatusCachePath },
+
+                // In Git 2.24.0, some new config settings were created. Disable them locally in VFS for Git repos in case a user has set them globally.
+                // https://github.com/microsoft/VFSForGit/pull/1594

```

Comment:
I would only have this comment once and have the 3 config settings it applies to follow it and have the comment be something like `The next 3 config settings were added in git 2.24.0. Disable ...`